tests/ci/azureml_tests/run_groupwise_pytest.py

```diff
@@ -2,90 +2,65 @@
 # Licensed under the MIT License.
 
 """
-run_pytest.py is the script submitted to Azure ML that runs pytest.
+run_groupwise_pytest.py is the script submitted to Azure ML that runs pytest.
 pytest runs all tests in the specified test folder unless parameters
 are set otherwise.
 """
 
-import sys
+import argparse
 import logging
 import pytest
-import argparse
-import glob
-import pkg_resources
-from azureml.core import Run
-from test_groups import nightly_test_groups, pr_gate_test_groups
-
-if __name__ == "__main__":
+import sys
 
-    logger = logging.getLogger("submit_groupwise_azureml_pytest.py")
-    logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+from test_groups import nightly_test_groups, pr_gate_test_groups
 
+def parse_args():
+    """
+    Parse command line arguments.
+    """
     parser = argparse.ArgumentParser(description="Process inputs")
+    parser.add_argument(
+        "--expname",
+        action="store",
+        default="persistentAzureML",
+        help="Experiment name on AzureML",
+    )
     parser.add_argument(
         "--testkind",
-        "-k",
         action="store",
         default="unit",
         help="Test kind - nightly or unit",
     )
     parser.add_argument(
         "--testgroup",
-        "-g",
         action="store",
         default="group_cpu_001",
         help="Group name for the tests",
     )
-    # Flag to indicate whether to turn off the warnings
     parser.add_argument(
         "--disable-warnings",
         action="store_true",
         help="Turn off warnings",
     )
-    args = parser.parse_args()
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+
+    logger = logging.getLogger("run_groupwise_pytest.py")
 
+    args = parse_args()
     if args.testkind == "nightly":
         test_group = nightly_test_groups[args.testgroup]
     else:
         test_group = pr_gate_test_groups[args.testgroup]
 
-    logger.info(f"Python version: {sys.version}")
-
-    logger.info("Installed packages:")
-    for p in pkg_resources.working_set:
-        logger.info(f" {p.project_name}:{p.version}")
-
-    logger.info("Tests to be executed")
-    logger.info(str(test_group))
-
-    # Run.get_context() is needed to save context as pytest causes corruption
-    # of env vars
-    run = Run.get_context()
-
-    logger.info("Executing tests now...")
-
     # Add options to pytest command (Duration and disable warnings)
     pytest_string = test_group + ["--durations"] + ["0"]
     if args.disable_warnings is True:
         pytest_string += ["--disable-warnings"]
 
     # Execute pytest command
-    pytest_exit_code = pytest.main(pytest_string)
-
+    logger.info("Executing tests now...")
+    sys.exit(pytest.main(pytest_string))
     logger.info("Test execution completed!")
```

Comment:
This throws a `SystemExit` exception, so the next one will never be executed, will it?

Reply:
You are right, haha